File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingMovementsResourceIntTest_jpaLoading.java
package uk.gov.justice.hmpps.prison.api.resource.impl;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.core.ParameterizedTypeReference;
import org.springframework.http.HttpMethod;
import uk.gov.justice.hmpps.prison.executablespecification.steps.AuthTokenHelper;

import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;

@AutoConfigureTestDatabase(replace = NONE)
public class BookingMovementsResourceIntTest_jpaLoading extends ResourceTest {

    private String token;

    @BeforeEach
    public void setup() {
        token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
    }

    @Test
    public void retrieveOffenderTransactionHistory_LazyLoads() {
        final var request = createHttpEntity(token, null);

        final var response = testRestTemplate.exchange(
                "/api/offenders/{offenderNo}/damage-obligations",
                HttpMethod.GET,
                request,
                new ParameterizedTypeReference<String>() {
                }, "A1234AA");

        assertThatStatus(response, 200);
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingMovementsResourceIntTest_jpaLoading.kt
@file:Suppress("ClassName")

package uk.gov.justice.hmpps.prison.api.resource.impl

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.boot.jdbc.test.autoconfigure.AutoConfigureTestDatabase
import org.springframework.core.ParameterizedTypeReference
import org.springframework.http.HttpMethod
import org.springframework.http.ResponseEntity
import uk.gov.justice.hmpps.prison.executablespecification.steps.AuthTokenHelper

@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class BookingMovementsResourceIntTest_jpaLoading : ResourceTest() {
  private lateinit var token: String

  @BeforeEach
  fun setup() {
    token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER)
  }

  @Test
  fun retrieveOffenderTransactionHistory_LazyLoads() {
    val request = createHttpEntity(token, null)

    val response: ResponseEntity<String> = testRestTemplate.exchange(
      "/api/offenders/{offenderNo}/damage-obligations",
      HttpMethod.GET,
      request,
      object : ParameterizedTypeReference<String>() {
      },
      "A1234AA",
    )

    assertThatStatus(response, 200)
  }
}
